frmts/tiledb/tiledbdataset.cpp

```diff
@@ -579,9 +468,51 @@ CPLErr TileDBRasterBand::IWriteBlock(int nBlockXOff, int nBlockYOff,
 
         if (status == tiledb::Query::Status::FAILED)
             return CE_Failure;
+        else
+            return CE_None;
     }
 
-    return CE_None;
+    return GDALPamRasterBand::IRasterIO(eRWFlag, nXOff, nYOff, nXSize, nYSize,
+                                        pData, nBufXSize, nBufYSize, eBufType,
+                                        nPixelSpace, nLineSpace, psExtraArg);
+}
+
+CPLErr TileDBRasterBand::IReadBlock(int nBlockXOff, int nBlockYOff,
+                                    void *pImage)
+{
+    const int nXOff = nBlockXOff * nBlockXSize;
+    const int nYOff = nBlockYOff * nBlockYSize;
+    const int nDTSize = GDALGetDataTypeSizeBytes(eDataType);
+    return IRasterIO(GF_Read, nXOff, nYOff, nBlockXSize, nBlockYSize, pImage,
```

Comment:
If nRasterXSize or nRasterYSize may not be multiple of the block size, then this call is going to be wrong for the right-most or bottom-most blocks, unless this is dealt in a special way  in IRasterIO().
Same for IWriteBlock()

Reply:
The array is padded to be have dimensions that are a multiple of the block size https://github.com/OSGeo/gdal/blob/e76b29a708f9cb5242e74e468ffc67a685466186/frmts/tiledb/tiledbdataset.cpp#L1922